clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -783,7 +784,13 @@ public void unsubscribe(TopicPartition... partitions) {
     private void scheduleAutoCommitTask(final long interval) {
         DelayedTask task = new DelayedTask() {
             public void run(long now) {
-                commit(CommitType.ASYNC);
+                commitAsync(new OffsetCommitCallback() {
```

Comment:
I thought about that too. Probably doesn't matter much and it's always kind of nice to have the code inline. As for using DefaultOffsetCommitCallback, either way is ok for me, though I doubt there's much chance of confusion between the auto and manual commits since users would (at least should) probably only be using one of them.